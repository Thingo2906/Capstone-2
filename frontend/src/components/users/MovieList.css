.movie-list {
  margin-left: 2%;
  padding-top: 5vh;
  height: 100vh; 
}

.poster{
  display: flex;
  overflow-y: scroll;
  flex-wrap: wrap;
  width: 100vw;
} 

.overview-text{
  color: black;
  text-align: left;
  border: 1px solid #ccc; /* Specify the border style and color */
  padding: 10px; /* Add some padding to the text within the border */
  display: block;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
  background-color: white;
  margin-top: 0;
}