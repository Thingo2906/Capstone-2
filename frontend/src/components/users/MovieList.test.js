import React from "react";
import { render } from "@testing-library/react";
import MovieList from "./MovieList";
import MovieApi from "../../api/api";
import UserContext from "../../auth/UserContext";

// Mock functions for required props
const mockAddMovieToList = jest.fn();
const mockRemoveMovieFromList = jest.fn();

// Mock UserContext with addedMovies property
const mockUserContext = {
  addedMovies: [], // we can customize this value as needed for the test
};

// Provide the mock UserContext as a context value

describe("MovieList Component", () => {
  it("renders without crashing", () => {
     render(
       <UserContext.Provider value={mockUserContext}>
         <MovieList
           fetchData={MovieApi.getAllMoviesFromList}
           addMovieToList={mockAddMovieToList}
           removeMovieFromList={mockRemoveMovieFromList}
         />
       </UserContext.Provider>
     );
  
  });

  it("renders the title", () => {
    const { asFragment, getByText } = render(
      <UserContext.Provider value={mockUserContext}>
        <MovieList
          fetchData={MovieApi.getAllMoviesFromList}
          addMovieToList={mockAddMovieToList}
          removeMovieFromList={mockRemoveMovieFromList}
        />
      </UserContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    const titleElement = getByText("Here is your list");
    expect(titleElement).toBeInTheDocument();
  });
});
