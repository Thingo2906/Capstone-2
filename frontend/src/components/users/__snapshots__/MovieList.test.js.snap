// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList Component renders the title 1`] = `
<DocumentFragment>
  <div
    class="movie-list"
  >
    <h2>
      Here is your list
    </h2>
    <div
      class="poster"
    />
  </div>
</DocumentFragment>
`;
